util/test/sub_test

```diff
@@ -1051,6 +1051,13 @@ for testname in testsrc:
     else:
       is_c_test = False
 
+    # the previous regex will not match multilocale C tests, so this check is
+    # mutually exclusive
+    if re.match(r'^.+\.ml-test\.c$', testname):
+        is_ml_c_test = True
+    else:
+        is_ml_c_test = False
+
```

Comment:
Maybe a little cleaner as:

```python
is_c_test = False
is_ml_c_test = False

if re.match(r'^.+\.test\.c$', testname):
    is_c_test = True
elif re.match(r'^.+\.ml-test\.c$', testname):
    is_ml_c_test = True
```
(or you could `is_c_test = re.match(r'^.+\.test\.c$', testname) != None` or whatever)

Oh actually, I'm not even sure why this is using a regex. Can this just be:

```python
is_c_test = testname.endswith(".test.c")
is_ml_c_test = testname.endswith(".ml-test.c")
```

?